src/OpenTelemetry.Exporter.Zipkin/Implementation/ZipkinActivityConversionExtensions.cs

```diff
@@ -0,0 +1,229 @@
+            {
+                remoteEndpoint = RemoteEndpointCache.GetOrAdd(attributeEnumerationState.RemoteEndpointServiceName, ZipkinEndpoint.Create);
+            }
+
+            var annotations = PooledList<ZipkinAnnotation>.Create();
+            ListEnumerator<ActivityEvent, PooledList<ZipkinAnnotation>>.AllocationFreeForEach(activity.Events, ref annotations, ProcessActivityEventsRef);
+
+            return new ZipkinSpan(
+                EncodeTraceId(context.TraceId, useShortTraceIds),
+                parentId,
+                EncodeSpanId(context.SpanId),
+                ToActivityKind(activity),
+                activity.OperationName,
+                activity.StartTimeUtc.ToEpochMicroseconds(),
+                duration: (long)activity.Duration.ToEpochMicroseconds(),
+                localEndpoint,
+                remoteEndpoint,
+                annotations,
+                attributeEnumerationState.Tags,
+                null,
+                null);
+        }
+
+        internal static string EncodeSpanId(ActivitySpanId spanId)
+        {
+            return spanId.ToHexString();
+        }
+
+        internal static long ToEpochMicroseconds(this DateTimeOffset dateTimeOffset)
+        {
+            return dateTimeOffset.Ticks / TicksPerMicrosecond;
+        }
+
+        internal static long ToEpochMicroseconds(this TimeSpan timeSpan)
+        {
+            return timeSpan.Ticks / TicksPerMicrosecond;
+        }
+
+        internal static long ToEpochMicroseconds(this DateTime utcDateTime)
+        {
+            const long UnixEpochTicks = 621355968000000000L; // = DateTimeOffset.FromUnixTimeMilliseconds(0).Ticks
+            const long UnixEpochMicroseconds = UnixEpochTicks / TicksPerMicrosecond;
+
+            // Truncate sub-microsecond precision before offsetting by the Unix Epoch to avoid
+            // the last digit being off by one for dates that result in negative Unix times
+            long microseconds = utcDateTime.Ticks / TicksPerMicrosecond;
+            return microseconds - UnixEpochMicroseconds;
+        }
+
+        private static string EncodeTraceId(ActivityTraceId traceId, bool useShortTraceIds)
+        {
+            var id = traceId.ToHexString();
+
+            if (id.Length > 16 && useShortTraceIds)
+            {
+                id = id.Substring(id.Length - 16, 16);
+            }
+
+            return id;
+        }
+
+        private static string ToActivityKind(Activity activity)
+        {
+            switch (activity.Kind)
+            {
+                case ActivityKind.Server:
+                    return "SERVER";
+                case ActivityKind.Producer:
+                    return "PRODUCER";
+                case ActivityKind.Consumer:
+                    return "CONSUMER";
+                case ActivityKind.Client:
```

Comment:
The Zipkin Span exporter defaults to "CLIENT", but here we default to "null".

Reply:
Yes, Zipkin Span exporter defaults to "CLIENT". There was an earlier comment from Reiley to handle ActivityKind.Internal.
Zipkin does not respect `Internal` kind. Java and Python OT exporter return Null for Internal. Borrowed the pattern from Java/Python to return NULL for Kind. Please let me know if we need to match the span exporter in this project.

Reply:
Ok this approach makes sense.